src/models/serverdata.cpp

```diff
@@ -21,56 +21,101 @@ bool ServerData::fromSettings() {
   SettingsHolder* settingsHolder = SettingsHolder::instance();
   Q_ASSERT(settingsHolder);
 
-  if (!settingsHolder->hasCurrentServerCountry() ||
-      !settingsHolder->hasCurrentServerCity() ||
-      !settingsHolder->hasCurrentServerCountryCode()) {
+  if (!settingsHolder->hasCurrentServerCountryCode() ||
+      !settingsHolder->hasCurrentServerCity()) {
     return false;
   }
 
   initializeInternal(settingsHolder->currentServerCountryCode(),
-                     settingsHolder->currentServerCountry(),
-                     settingsHolder->currentServerCity());
+                     settingsHolder->currentServerCity(),
+                     settingsHolder->entryServerCountryCode(),
+                     settingsHolder->entryServerCity());
 
-  logger.debug() << m_countryCode << m_countryName << m_cityName;
+  logger.debug() << toString();
+  return true;
+}
+
+bool ServerData::fromString(const QString& data) {
```

Comment:
This is just for testing, right?

Reply:
Can we expose it only `#ifdef UNIT_TEST` ?

Reply:
I'll make a note to myself to remove this function if we don't wind up using it after merging with #1603